src/librustc_typeck/outlives/explicit.rs

```diff
@@ -33,9 +33,9 @@ impl<'tcx> ExplicitPredicatesMap<'tcx> {
     ) -> &RequiredPredicates<'tcx> {
         self.map.entry(def_id).or_insert_with(|| {
             let predicates = if def_id.is_local() {
-                tcx.explicit_predicates_of(def_id).predicates
+                tcx.explicit_predicates_of(def_id).predicates.clone()
             } else {
-                tcx.predicates_of(def_id).predicates
+                tcx.predicates_of(def_id).predicates.clone()
```

Comment:
Unnecessary clone, can just keep the whole `Lrc` in the `predicates` variable and change the loop below.

Reply:
I tried and failed to do that. I can't use the `Rc` directly, because that would move it. I can't use a reference, because it doesn't live long enough.

Reply:
Ok, I worked out a different way to do it.